### DISCUSSION TITLE
Сортировка при выборке по значению селекта, а не его номеру

### DISCUSSION THREAD

**Discussion by juramebel** [totumonline/totum-ru-issues-and-discussions#55] [Как это сделать?] [answered]
Здравствуйте!
Такая задача, помогите решить) 
Есть временная таблица, в ней поле h_insert.
Оно набирается из циклов через ярлыки. 
Важно, чтобы данные в h_insert всегда имели сортировку по полю, например, заказы.
А это поле - селект. 
И id их не совпадают с номерами.
Как сделать сортировку при выборке через selectList по значению селекта в исходной таблице, а не по его номеру?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Cамый топорный способ — это в таблице из которой делается выборка сделать техническое поле (строка), которое через `#s.select_field` будет брать его отображение и сортировку в `select` потом делать по нему.

Для ярлыков это тоже работает.

Иногда бывает, что вариант добавления поля не годится. Поэтому можно пойти так. Взять `rowlist` и вместо `field: 'select_field'` использовать `sfield: 'select_field'`. И потом этот `rowlist` отсортировать:

```
= : listSort(list: $rowlist; type: "string"; direction: "asc"; key: "select_field")

		rowlist: selectRowList(table: 'table'; sfield: 'select_field'; field: 'some_data')

```

Первый вариант добавляет небольшой оверхед при выполнении любого действия в цикле. Второй — оверхед при выполнении расчета h_insert тк для каждой строки будет рассчитан код-селекта поля указанного в `sfield`.

**Reply by juramebel** [reply]
Второй выглядит изысканнее, но сделал по первому варианту)
Спасибо!
